Documentation for SOS (#22117)
dotnet/docs: docs/core/diagnostics/dumps.md

## Analyze dumps

> Dumps can be analyzed using the [`dotnet-dump`](dotnet-dump.md) CLI tool or with [Visual Studio](https://docs.microsoft.com/visualstudio/debugger/using-dump-files).

Reviewer: ```suggestion
You can analyze dumps using the [`dotnet-dump`](dotnet-dump.md) CLI tool or with [Visual Studio](/visualstudio/debugger/using-dump-files).
```

> > [!NOTE] Visual Studio 16.8 and later allows you to [open Linux dumps](https://devblogs.microsoft.com/visualstudio/linux-managed-memory-dump-debugging/) generated on .NET Core 3.1.7 or later.  

Reviewer: ```suggestion
> [!NOTE]
> Visual Studio version 16.8 and later allows you to [open Linux dumps](https://devblogs.microsoft.com/visualstudio/linux-managed-memory-dump-debugging/) generated on .NET Core 3.1.7 or later.  
```

> > [!NOTE] If native debugging is necessary, the [SOS debugger extension](sos-debugging-extension.md) can be used with and LLDB on Linux/MacOS. SOS is also supported with [Windbg/cdb](https://docs.microsoft.com/windows-hardware/drivers/debugger/debugger-download-tools) on Windows though Visual Studio is recommended.

Reviewer: ```suggestion
> [!NOTE]
> If native debugging is necessary, you can use the [SOS debugger extension](sos-debugging-extension.md) with LLDB on Linux and macOS. SOS is also supported with [Windbg/cdb](/windows-hardware/drivers/debugger/debugger-download-tools) on Windows, although Visual Studio is recommended.
```

## Analyze dumps

> Dumps can be analyzed using [`dotnet-dump`](dotnet-dump.md), or the [SOS debugger extension](sos-debugging-extension.md) with [Windbg/cdb](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/debugger-download-tools) (Windows) and LLDB (Linux/MacOS).

Reviewer 2: I am a bit scared/intimidated by these latter tools.  So I would question if we want to direct our customers to these more advanced tools.

Do we want to help them chose here?

> Dumps can be analyzed using [`dotnet-dump`](dotnet-dump.md), or the [SOS debugger extension](sos-debugging-extension.md) with [Windbg/cdb](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/debugger-download-tools) (Windows) and LLDB (Linux/MacOS).

Reviewer 3: ```suggestion
Dumps can be analyzed using [`dotnet-dump`](dotnet-dump.md), or the [SOS debugger extension](sos-debugging-extension.md) with [Windbg/cdb](https://docs.microsoft.com/windows-hardware/drivers/debugger/debugger-download-tools) (Windows) and LLDB (Linux/MacOS).
```
